### PR TITLE
Made front page translatable

### LINKED ISSUE #5104: Front page is not translatable
See https://github.com/HabitRPG/habitrpg/blob/develop/website/views/static/front.jade most strings are plain text, we need to move them to locales files.

### PR DESCRIPTION
**Pull request by NotGonnaGitUs**
Made front page translatable. Fixes #5104

### CODE REVIEW

**Comment by paglias** [comment]
@NotGonnaGitUs great, thanks! 

Just one thing, if it's not too much work, you should remove the old strings that are not used anymore from `common/locales/en/front.json`

**Comment by paglias** [comment]
That file is only used for the static pages, so what's not used there can be removed

**Comment by NotGonnaGitUs** [comment]
Oh. I did not know that the file is only used for front.jade. 
Thanks. I'll clean it up and send another pull request.

**Comment by NotGonnaGitUs** [comment]
While I was cleaning it up, I found that I could not find the environment values for "groupPlans" and "contactUs" in the json (https://github.com/HabitRPG/habitrpg/blob/develop/common/locales/en/front.json), whereas the same are used in the jade file(https://github.com/HabitRPG/habitrpg/blob/develop/website/views/static/front.jade) and appear fine. 

Am I missing something @paglias ? Does the front.jade uses other json files too?

**Comment by paglias** [comment]
So front.jade can use all json files but we try to keep them separate which
means that strings in front.json should be used only for front.jade
Il 02/mag/2015 13:07, "NotGonnaGitUs" notifications@github.com ha scritto:

**Comment by NotGonnaGitUs** [comment]
Okay. So I'll make front.json contain only the objects used in front.jade and do a new pull request.

**Comment by NotGonnaGitUs** [comment]
Is there anything else I can improve on here @paglias ?

Edit: Nevermind - I think I deleted a few extra objects (being used by footer). Working on it.

**Comment by NotGonnaGitUs** [comment]
Okay. I think the JSON is done. The test fails due to the recent login change (I think) I'll try rebasing local branch and doing the push again.

**Comment by crookedneighbor** [comment]
Looks good except for this:

screen shot 2015-05-02 at 9 08 15 am

File: common/locales/en/front.json [hunk 1]
```diff
@@ -1,100 +1,110 @@
...
-    "marketing4Lead3-1": "Want to gamify your life?",
-    "marketing4Lead3-2": "Interested in running a group in education, wellness, and more?",
-    "marketing4Lead3-3": "Want to learn more?",
-
-    "playButton": "Play",
-    "username": "Username",
-    "password": "Password",
-    "useUUID": "Use UUID / API Token (For Facebook Users)",
-    "passMan": "In case you are using a password manager (like 1Password) and have problems logging in, try typing username and password manually.",
-    "forgotPass": "Forgot Password",
```
Comment (crookedneighbor) [inline common/locales/en/front.json]: Still used.

**Comment by crookedneighbor** [comment]
It's failing because you removed the login string: https://github.com/HabitRPG/habitrpg/blob/develop/common/locales/en/front.json#L67

**Comment by NotGonnaGitUs** [comment]
Ah. I was missing the uses by login modal. Let me work on it.

**Comment by crookedneighbor** [comment]
Good work!

I'd like to write a script to cycle through the removed strings and check if they're used elsewhere in the repository before this gets merged in. I'm pretty sure they're not used elsewhere, but I'd hate to be wrong.

**Comment by NotGonnaGitUs** [comment]
Thanks! Can I help in that?

**Comment by crookedneighbor** [comment]
Looks like there are some strings that were deleted that were being used on the static subpages. Here's a list of all the places they're used:
- [ ] website/views/static/press-kit.jade:18:  p= env.t('presskitText')
- [ ] website/views/static/press-kit.jade:19:  p= env.t('presskitDownload')
- [ ] website/views/static/features.jade:89:              a.btn.btn-primary(href='/static/videos')=env.t('watchVideos')
- [ ] website/views/static/layout.jade:48:              a(href='/static/videos')=env.t('companyVideos')
- [ ] website/views/static/videos.jade:8:  title=env.t('companyVideos')
- [ ] website/public/js/controllers/rootCtrl.js:192:        title: window.env.t('history'),
- [ ] website/views/shared/header/menu.jade:16:                a(ui-sref='tasks')=env.t('tasks')
- [ ] website/views/shared/header/menu.jade:80:          span=env.t('tasks')
- [ ] website/views/shared/header/menu.jade:237:                a(ng-click='logout()')=env.t('logout')
- [ ] website/views/shared/header/menu.jade:227:        a(ng-click='User.sync()', popover=env.t('sync'),popover-placement='bottom',popover-trigger='mouseenter')
- [ ] website/views/shared/header/menu.jade:255:                a(href="https://vimeo.com/57654086", target='_blank')=env.t('tutorials')
- [ ] website/views/shared/header/menu.jade:278:        .npc_bailey.npc_bailey_head(popover=env.t('psst'), popover-trigger='mouseenter', popover-placement='right', ng-click='openModal("newStuff",{size:"lg"})')
- [ ] website/views/static/features.jade:93:        p.lead=env.t('landingp1')
- [ ] website/views/static/features.jade:94:        h2=env.t('landingp2header')
- [ ] website/views/static/features.jade:98:          =env.t('landingp2')
- [ ] website/views/static/features.jade: 100:        h2=env.t('landingp3header')
- [ ] website/views/static/features.jade:103:          =env.t('landingp3')
- [ ] website/views/static/features.jade:105:        h2=env.t('landingp4header')
- [ ] website/views/static/features.jade:106:        p.lead=env.t('landingp4')
- [ ] website/views/static/features.jade:108:          h2=env.t('landingend')
- [ ] website/views/static/features.jade:110:            =env.t('landingend2')
- [ ] website/views/static/features.jade:111:            a(href="FEATURESPAGEHERE")=env.t('landingfeatureslink')
- [ ] website/views/static/features.jade:112:            =env.t('landingend3')
- [ ] website/views/static/features.jade:113:            a(href="ENTERPRISEPAGEHERE")=env.t('landingadminlink')
- [ ] website/views/static/features.jade:115:            =env.t('landingend4')
- [ ] website/views/static/features.jade:14:        h1=env.t('marketing1Header')
- [ ] website/views/static/features.jade:17:            a.gallery(href='/marketing/screenshot.png', title=env.t('marketing1Header'))
- [ ] website/views/static/features.jade:19:            p.lead=env.t('marketing1Lead1')
- [ ] website/views/static/features.jade:24:            p.lead!=env.t('marketing1Lead2')
- [ ] website/views/static/features.jade:22:            a.gallery(href='/marketing/gear.png', title=env.t('marketing1Lead2Title'))
- [ ] website/views/static/features.jade:28:            p.lead!=env.t('marketing1Lead3')
- [ ] website/views/static/features.jade:26:            a.gallery(href='/marketing/drops.png', title=env.t('marketing1Lead3Title'))
- [ ] website/views/static/features.jade:34:        h1=env.t('marketing2Header')
- [ ] website/views/static/features.jade:37:            a.gallery(href='/marketing/guild.png', title=env.t('marketing2Header'))
- [ ] website/views/static/features.jade:39:            p.lead=env.t('marketing2Lead1')
- [ ] website/views/static/features.jade:43:            p.lead!=env.t('marketing2Lead2')
- [ ] website/views/static/features.jade:41:            a.gallery(href='/common/img/sprites/spritesmith/quests/quest_vice3.png', title=env.t('marketing2Lead2Title'))
- [ ] website/views/static/features.jade:47:            p.lead!=env.t('marketing2Lead3')
- [ ] website/views/static/features.jade:51:        h1=env.t('marketing3Header')
- [ ] website/views/static/features.jade:56:            p.lead!=env.t('marketing3Lead1')
- [ ] website/views/static/features.jade:60:            p.lead!=env.t('marketing3Lead2')
- [ ] website/views/static/features.jade:64:        h1=env.t('marketing4Header')
- [ ] website/views/static/features.jade:67:            h3=env.t('marketing4Lead1Title')
- [ ] website/views/static/features.jade:69:            p.lead=env.t('marketing4Lead1')
- [ ] website/views/static/features.jade:71:            h3=env.t('marketing4Lead2Title')
- [ ] website/views/static/features.jade:73:            p.lead=env.t('marketing4Lead2')
- [ ] website/views/static/features.jade:76:            h3=env.t('marketing4Lead3Title')
- [ ] website/views/static/features.jade:79:              =env.t('marketing4Lead3-1')
- [ ] website/views/static/features.jade:83:              =env.t('marketing4Lead3-2')
- [ ] website/views/static/features.jade:87:              =env.t('marketing4Lead3-3')
- [ ] website/views/static/features.jade:81:              a.btn.btn-primary(ng-click='playButtonClick()')=env.t('playButton')
- [ ] website/views/static/layout.jade:58:          button#header-play-button.btn.btn-primary.navbar-btn.navbar-right(ng-click='playButtonClick()')=env.t('playButton')
- [ ] website/public/js/controllers/authCtrl.js:71:          alert(window.env.t('invalidEmail'));

**Comment by NotGonnaGitUs** [comment]
Yep. I never looked in those places. Here are the steps I followed for a more comprehensive search- 
1. Used grep to find all uses of env.t
2. Regex extracted the parameters passed to env.t
3. Compared each key of front.json to see if in list above
4. Made new JSON
5. The removed keys i.e. not used anywhere with env.t are  - [anonymous,contact,home,learnMore,marketing3Lead1Title,options,tagline,titleFront]
6. I was surprised at contact being included here but I guess it has been replaced by contactUs everwhere.
Please let me know if I missed anything.

**Comment by crookedneighbor** [comment]
Looks good to me now. :+1:

**Comment by crookedneighbor** [comment]
This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by NotGonnaGitUs** [comment]
Awesome! 
Is there something else I can help with? The other entry level coding tasks were started like a year ago. Are they still relevant?

**Comment by crookedneighbor** [comment]
Tier awarded!

This one: https://github.com/HabitRPG/habitrpg/issues/4522 is still very much relevant. There are still parts of the site that are not translatable.

This one is also relevant: https://github.com/HabitRPG/habitrpg/issues/2094

The others are also relevant, but still in discussion, so I hesitate to say work on those, but if any look interesting to you, be sure to chime in with your opinion.

**Comment by NotGonnaGitUs** [comment]
Thanks! Woo hoo. 

I'll start with #4522  since I have experience in that and read more about #2094  before starting on it.

**Pull request merged**